src/Resources/config/services.php

```diff
@@ -2,29 +2,23 @@
 
 declare(strict_types=1);
 
-/*
- * The MIT License (MIT)
- *
- * Copyright (c) 2014-2020 Spomky-Labs
- *
- * This software may be modified and distributed under the terms
- * of the MIT license.  See the LICENSE file for details.
- */
-
 use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
 use Yoanbernabeu\AirtableClientBundle\AirtableClient;
+use Yoanbernabeu\AirtableClientBundle\AirtableClientInterface;
 
 return static function (ContainerConfigurator $container): void {
     $container = $container->services()->defaults()
-        ->private()
+        ->public()
         ->autoconfigure()
         ->autowire()
     ;
 
-    $container->set(AirtableClient::class)
+    $container->set(AirtableClient::class, AirtableClient::class)
         ->args([
             '%yoanbernabeu_airtable_client.airtable_client.key%',
             '%yoanbernabeu_airtable_client.airtable_client.id%',
         ])
     ;
+
+    $container->alias(AirtableClientInterface::class, AirtableClient::class);
```

Comment:
:bulb: `$container` is fluent,
```php
$container->set()
->alias()
;
```

IMO having a "key" instead of FQCN as an alias is more easier to manipulate for the end user (definition, decoration...)
`->alias(AirtableClientInterface::class, 'airtable_client')` but don't forget to create `airtable_client`

Reply:
```php
$container->set('airtable_client', AirtableClient::class)
        ->args([
            '%yoanbernabeu_airtable_client.airtable_client.key%',
            '%yoanbernabeu_airtable_client.airtable_client.id%',
            service(HttpClientInterface::class),
            service(ObjectNormalizer::class),
        ])
    ;

    $container->alias(AirtableClientInterface::class, AirtableClient::class);
```

Is not the same that : 
```php
    $container->set('airtable_client', AirtableClient::class)
        ->args([
            '%yoanbernabeu_airtable_client.airtable_client.key%',
            '%yoanbernabeu_airtable_client.airtable_client.id%',
            service(HttpClientInterface::class),
            service(ObjectNormalizer::class),
        ])
        ->alias(AirtableClientInterface::class, AirtableClient::class)
    ;
```

The object return by the method `args` is not the `$container`

Reply:
I let the code of rebase #5, I don't even seen this 2 lines of code, I'll delete them and we'll see

Reply:
What is the result? I had in mind that either we don't define the arguments (and then autowiring is used) or we have to define everything.
I may be wrong, but because of the presence of the 2 params, other arguments shall be set.

Reply:
`args` comes from `ArgumentTrait` which is used by `ServiceConfigurator` so it should works too but it's also OK like you did :sweat_smile:

Reply:
It works with and without the references ;)

Reply:
>I may be wrong, but because of the presence of the 2 params, other arguments shall be set.

Not sure about this, I think missing params will be autowired

Reply:
> `args` comes from `ArgumentTrait` which is used by `ServiceConfigurator` so it should works too but it's also OK like you did 😅

Yes it's 2 differents instance of `ServiceConfigurator`, but that make any differences, you right, it works too, so I change it for a fluent one

Reply:
Can we resolve this conversation ? @ismail1432